spec/requests/products/show_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "web_spec"
+
+RSpec.describe "GET /products/:id", type: :request do
+  context "with a product that doesn't exist" do
+    before do
+      do_request(100)
```

Comment:
It is almost impossible to recognize what `100` is for. It should be rather a keyword argument or a hash so `do_request(id: 100)` in this case. https://robots.thoughtbot.com/ruby-2-keyword-arguments